lib/ansible/modules/cloud/vmware/vmware_vm_sp.py

```diff
@@ -0,0 +1,243 @@
+
+# Prasanna Nanda <pnanda@cloudsimple.com>
+
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: vmware_vm_sp
+short_description: VM Storage Profile Utilities
+description:
+    - Attach/Detach a Storage Profile to a VM
+version_added: 2.4
+author:
+    - Prasanna Nanda <pnanda@cloudsimple.com>
+notes:
+    - Tested on vSphere 6.5
+requirements:
+    - "python >= 2.7"
+    - PyVmomi
+options:
+    vmname:
+        description:
+            - Name of the VM to which Storage Profile will be attached/detached/check
+        required: True
+    profilename:
+        description:
+            - Name of the Storage Profile which will be attached/detached
+        required: True
+    action:
+        description:
+            - Valid values are attach/detach/check
+        required: True
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = '''
+  - name: Attach/Detach a VM to a Storage Profile or Check Compliance
+    vmware_vm_sp:
+      hostname: A vCenter host
+      username: vCenter username
+      password: vCenter password
+      vmname: Name of the VM to which Storage Profile will be attached/detached/check
+      profilename: Name of the Storage Profile which will be attached/detached/check
+      action: attach|detach|check
+'''
+
+RETURN = ''' # '''
+
+try:
+    from pyVmomi import vim, pbm, vmodl
+    from pyVim.connect import SoapStubAdapter
+    from pyVim.connect import SmartConnect, Disconnect
+    HAS_PYVMOMI = True
+except ImportError:
+    HAS_PYVMOMI = False
+
+import time
+from ansible.module_utils.vmware import vmware_argument_spec, connect_to_api
+import ssl
+import atexit
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.spbm_client import SPBMClient
+
+
+class vmware_vm_sp(object):
+    def __init__(self, module):
+        self.module = module
+        self.vmname = module.params['vmname']
+        self.profilename = module.params['profilename']
+        self.action = module.params['action']
+        self.profile = None
+        try:
+            context = None
+            context = ssl._create_unverified_context()
+            si = SmartConnect(host=module.params['hostname'],
```

Comment:
Why I could not use connect_to_api of vmware.py, because SPBM needs to
have access to ServiceInstance for Stub Generation. connect_to_api returns
a ServiceContent